### ISSUE TITLE
Remove Context and simplify LpActionCodeEscaper.kt

### ISSUE THREAD

**Issue by dblanovschi** [Mcat12/intellij-lalrpop#30]
First of all, happy new year!
I'd like to discuss [this change](https://github.com/dblanovschi/intellij-lalrpop/commit/1dfe583f03776e9c8e5b8a29f2e4c6208c13b502#diff-1d6098cca281aaa5c4e970a3736b9f6cd81b3dba5f3ac99542e6f90a5f28a449L139-L149), which came after a realization that the whole `Context` enum might have been useless(we have [field initialization shorthand](https://doc.rust-lang.org/edition-guide/rust-2018/data-types/field-init-shorthand.html) after all), and the whole commit simplifies [`LpActionCodeEscaper.kt`](https://github.com/Mcat12/intellij-lalrpop/blob/master/src/main/kotlin/com/mdrobnak/lalrpop/psi/ext/LpActionLiteralTextEscaper.kt) by removing it. It looks like the only valid usage of `{a:a, b:b, c:c}` where expanding `{<>}` to `{a, b, c}` will fail in the entire rust grammar is in a struct declaration, e.g.:

```rust
// Expanding
struct Something {<>}
// to the invalid
struct Something {a, b, c}
// rather than the valid
struct Something {a: a, b: b, c: c}
```
Though it seems ***very very*** unlikely anyone would use `<>` in a struct declaration, and I do hope nobody uses something like this.

I also tried to [change this](https://github.com/lalrpop/lalrpop/blob/ff5194452e00f7077d5f76166e691bbbc6e7177b/lalrpop/src/normalize/lower/mod.rs#L399-L408) from lalrpop's sources to use the field init shorthand and ran all of lalrpop's tests; the only failing one was `verify_lalrpop_generates_itself` but it would fail even without the change(no idea why).

So @Mcat12 what do you think?

Edit: someone mentioned macros and they would indeed make for very weird cases

**Comment by Mcat12** [comment]
Happy new year! I think it would be worthwhile to make an issue on lalrpop's repo to get more context on why they don't use field init shorthand (maybe it's just a legacy thing), and what impact it might have if they switch to the shorthand.

I think it's fine for us as an IDE plugin to experiment and make this change. It's pretty easy to roll back. It will probably help inform lalrpop of issues that might might arise if they made the change. The fact that the lalrpop test results didn't change after making the switch is encouraging.

**Comment by dblanovschi** [comment]
Created https://github.com/lalrpop/lalrpop/issues/579

**Comment by Mcat12** [comment]
Closed by #42
